### ISSUE TITLE
Disambiguate approach vs `pulumi preview --import-file`

### ISSUE THREAD

**Issue by t0yv0** [pulumi/pulumi-tool-cdk-importer#3]
The code in pulumi-tool-cdk-importer as of Dec 3 was built without awareness of https://github.com/pulumi/pulumi/issues/14510 work @justinvp and @frassle carried out, in particular note the blog post:

https://github.com/pulumi/pulumi-hugo/pull/3865/files#diff-29ebc9f1dce19570d9affcc79b93dcaa658e8a426d06e532bc6ee569440b0fbcR105

Apparently Pulumi can now do:

```
pulumi preview --import-file my-import-file.json 
pulumi import --file my-import-file.json 
```

Not sure if `pulumi import` can skip code generation but looks like that should be doable, currently got:

```
      --generate-code                         Generate resource declaration code for the imported resources (default true)
```

There's some additional changes that allow component resources to participate in import.

We might want to have another look at seeing if we can unify the CDK import scenario with this approach so that this tool becomes redundant and we bake any necessary functionality into pulumi-cdk itself, so Pulumi CLI can import the CDK states directly. 

To recap, the scenario this tool addresses is:

1. user starts from CDK sources and CDK->CF provisioned state in the cloud
2. user rewrites CDK sources to pulumi-cdk
3. user wants to get the import pulumi-cdk state that'd result in empty preview (not the program, as that's at hand)

The way this is currently addressed in this tool:

1. run `pulumi up` against a proxied `pulumi-aws-native` provider
2. component resources create naturally
3. the proxied aws-native provider rewrites `Create` calls to `Read` to fetch the state from CloudControl API (import emulation)
4. the tool does adaptation between CDK<->CloudControl resource IDs which is the non-trivial bit

CC @corymhall
